magprime/automated_emails.py

```diff
@@ -57,27 +57,24 @@ def __init__(self, event):
 StopsEmail('MAGFest Staff Support', 'staff_support.txt',
            lambda a: a.assigned_to(c.STAFF_SUPPORT) and not a.trusted_in(c.STAFF_SUPPORT))
 
-MarketplaceEmail('Your MAGFest Marketplace Application', 'marketplace_delay.txt',
-                 lambda g: g.status == c.UNAPPROVED and g.registered < datetime(2015, 10, 29, tzinfo=c.EVENT_TIMEZONE))
-
 # Left here in case the decision to remove is re-visited later. 20160809 RAE
 # MarketplaceEmail('Your MAGFest Marketplace Application has been waitlisted', 'marketplace_auto_waitlisted.txt',
 #                 lambda g: g.status == c.WAITLISTED and g.registered >= c.DEALER_REG_DEADLINE)
 
 StopsEmail('MAGFest Dept Checklist Introduction', 'dept_checklist_intro.txt',
            lambda a: a.is_single_dept_head and a.admin_account)
 
-AutomatedEmail(Attendee, 'Last Chance for MAGFest 2016 bonus swag!', 'attendee_swag_promo.html',
+AutomatedEmail(Attendee, 'Last Chance for MAGFest ' c.YEAR ' bonus swag!', 'attendee_swag_promo.html',
                lambda a: a.can_spam and
                          (a.paid == c.HAS_PAID or a.paid == c.NEED_NOT_PAY or (a.group and a.group.amount_paid)) and
                          days_after(3, a.registered) and
                          days_before(14, c.SUPPORTER_DEADLINE))
 
 # Send to any attendee who will be receiving a t-shirt (staff, volunteers, anyone
 # who kicked in at the shirt level or above).	Should not be sent after the t-shirt
-# size deadline (a new deadline not yet recorded in uber).
-AutomatedEmail(Attendee, 'MAGFest 2016 t-shirt size confirmation', 'confirm_shirt_size.html',
-               lambda a: before(datetime(2016, 1, 16, tzinfo=c.EVENT_TIMEZONE)) and
+# size deadline.
+AutomatedEmail(Attendee, 'MAGFest ' c.YEAR ' t-shirt size confirmation', 'confirm_shirt_size.html',
+               lambda a: before(c.SHIRT_DEADLINE) and
```

Comment:
Hey, I just found the deadline in the YAML, so I used it. I'm a bit
concerned that the code I replaced used the TZ, but I wasn't sure how to do
so with this.

--
Robert Earl <r.a.earl@gmail.com>
FAX: 1-440-919-5086

On Tue, Aug 9, 2016 at 7:16 AM, Dominic Cerquetti <notifications@github.com>
wrote:

> In magprime/automated_emails.py
> <https://github.com/magfest/magprime/pull/99#discussion_r74039186>:
>
> >                 lambda a: a.can_spam and
> >                           (a.paid == c.HAS_PAID or a.paid == c.NEED_NOT_PAY or (a.group and a.group.amount_paid)) and
> >                           days_after(3, a.registered) and
> >                           days_before(14, c.SUPPORTER_DEADLINE))
> >
> >  # Send to any attendee who will be receiving a t-shirt (staff, volunteers, anyone
> >  # who kicked in at the shirt level or above).	Should not be sent after the t-shirt
> > -# size deadline (a new deadline not yet recorded in uber).
> > -AutomatedEmail(Attendee, 'MAGFest 2016 t-shirt size confirmation', 'confirm_shirt_size.html',
> > -               lambda a: before(datetime(2016, 1, 16, tzinfo=c.EVENT_TIMEZONE)) and
> > +# size deadline.
> > +AutomatedEmail(Attendee, 'MAGFest ' c.YEAR ' t-shirt size confirmation', 'confirm_shirt_size.html',
> > +               lambda a: before(c.SHIRT_DEADLINE) and
>
> +1 on c.SHIRT_DEADLINE
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/magfest/magprime/pull/99/files/9e5f6f3375afae9bb2efa2604155c34e49d86e75#r74039186>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIXSFZj9VmOXoQzOraV3wntzcQtZPelJks5qeGGTgaJpZM4Jf818>
> .
>